internal/cache/config.go

```diff
@@ -0,0 +1,71 @@
+package cache
+
+import (
+	"context"
+	"time"
+
+	"github.com/zitadel/logging"
+)
+
+type Config struct {
+	// Name used to refer to this cache.
+	// May be used for logging or storage specific needs,
+	// like a table name.
+	Name string
+
+	// Age since an object was added to the cache,
+	// after which the object is considered invalid.
+	// 0 disables max age checks.
+	MaxAge time.Duration
+
+	// Age since last use (Get) of an object,
+	// after which the object is considered invalid.
+	// 0 disables last use age checks.
+	LastUseAge time.Duration
+
+	// Interval at which the cache is automatically pruned.
+	// 0 disables automatic pruning.
+	AutoPruneInterval time.Duration
+
+	// Timeout for an automatic prune.
+	// It is recommended to keep the value shorter than AutoPruneInterval
+	// 0 disables timeouts.
+	AutoPruneTimeOut time.Duration
```

Comment:
I've re-arranged it the following way:

1. `Prune()` is removed from the `Cache` interface
2. `Pruner()` is an optional interface to `Cache`
3. Added a seperate `AutoPrunerConfig`

For now I've removed the Auto prune start from the `gomap` implementation, because I'm not sure yet how I will map configuration to `defaults.yaml` after.